lib/galaxy/tools/__init__.py

```diff
@@ -934,11 +935,9 @@ def _repository_dir(self):
 
         if getattr(self, 'tool_shed', None):
             tool_dir = Path(self.tool_dir)
-            if tool_dir.parts[-1] == self.repository_name:
-                return str(tool_dir)
-            for parent in tool_dir.parents:
-                if parent.parts[-1] == self.repository_name:
-                    return str(parent)
+            for repo_dir in itertools.chain([tool_dir], tool_dir.parents):
+                if repo_dir.stem == self.repository_name:
```

Comment:
@dannon @mvdbeek What about changing `repo_dir.stem` to `repo_dir.name` ? That should be correct even in the unlikely case that `self.repository_name` contains a `.`